finder-app/writer.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Author: Sudarshan J
+ * Filename: writer.c
+ * File Description: Replacement for the writer.sh script to create a file and a string to it.
+ * Reference: Linux SP by Robert Love Chapter 2, Man pages.
+*/
+
+//Includes
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <syslog.h>
+
+//Prototypes
+static void display_msg(void);
+
+
+int main(int argc, char **argv)
+{
+	display_msg();
+
+	//Check corner case for number of arguments as in writer.sh
+	
+	if(argc != 3)
+	{
+	printf("Error! Number of arguments need to be executable + 2\n");	       printf("Correct format: ./writer <filepath> <text to write>\n");
+	return EXIT_FAILURE;  
+	}
+
+	//File Descriptor var
+	int file_descriptor;
+
+	//Variable to store bytes that are written
+	ssize_t bytes_writ;
+
+	//Filepath and string to be written
+	char *filepath = argv[1];
+	char *string = argv[2];
```

Comment:
@suja8742 I like the assignment to variables.  See discussion about const correct from class today about an additional improvement here.